src/BattleState.cpp

```diff
@@ -67,27 +66,21 @@ BattleState::BattleState(string stage, string cmusic, vector< pair<string, strin
 }
 
 BattleState::~BattleState(){
-//	for(unsigned i = 0; i < fighters.size(); i++)
-//		delete(fighters[i]);
-
-//	fighters.clear();
 }
 
 void BattleState::update(float delta){
 	InputManager * input_manager = InputManager::get_instance();
 
-	if(input_manager->key_press(SDLK_ESCAPE) or input_manager->joystick_button_press(InputManager::SELECT, 0)){
+	if(input_manager->quit_requested() ||
+		input_manager->key_press(InputManager::K_SELECT) ||
+		input_manager->joystick_button_press(InputManager::SELECT, 0)
+	){
 		music.stop();
 		m_quit_requested = true;
 		Game::get_instance().push(new MenuState());
 		return;
 	}
 
-	if(input_manager->quit_requested()){
```

Comment:
This if should not be removed, it closes the window when pressed Alt+F4 or when clicked the close window button. The other way, it returns to the menu, which is wrong